### PR TITLE
Update help command to accept parameters

### LINKED ISSUE #124: Bug with URL validation regex
Currently, our regex for valid URL validation is 
``` public static final String VALIDATION_REGEX = "\\b(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]";```

However, this does not catch some invalid URLs, once such example is ```https://..com/```

### PR DESCRIPTION
**Pull request by tanjiajiajun**
Resolves #124, #139

- Add HelpCommandParser to parse Help command Arguments
- Allow Help Command to accept Arguments
- Add JUnit Tests for HelpCommand and HelpCommandParser
- Add JUnit Tests for ParserUtil, AddCommandParser and EditCommandParser for Links
- Update UG and DG

### CODE REVIEW

File: src/test/java/educonnect/logic/commands/HelpCommandTest.java [hunk 1]
```diff
@@ -2,19 +2,30 @@
...
         assertCommandSuccess(new HelpCommand(), model, expectedCommandResult, expectedModel);
     }
+
+    @Test
+    public void execute_helpValidArgs_success() {
+        String validArg = VALID_COMMANDS[3];
+        CommandResult expectedCommandResult = new CommandResult(SHOWING_HELP_MESSAGE_EDIT);
+        assertCommandSuccess(new HelpCommand(validArg), model, expectedCommandResult, expectedModel);
+    }
 }
```
Comment (fy17ohhh) [inline src/test/java/educonnect/logic/commands/HelpCommandTest.java:31]: good test cases:)

File: src/main/java/educonnect/model/student/Link.java [hunk 2]
```diff
@@ -9,8 +9,12 @@
 public class Link {
 
     public static final String MESSAGE_CONSTRAINTS = "Please provide a valid website.";
+
+    //    This validation regex is obtained from Mustofa Rizwan at
+    //    https://stackoverflow.com/questions/42618872/regex-for-website-or-url-validation
```
Comment (fy17ohhh) [inline src/main/java/educonnect/model/student/Link.java:14]: For reused work could follow this guideline
https://nus-cs2103-ay2324s2.github.io/website/admin/appendixB-policies.html#giving-credit-for-reused-work

File: src/test/java/educonnect/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -321,6 +324,28 @@ public void parseTimetable_validInputs_returnsCompletedTimetable() throws Except
...
+        Link expectedLink = new Link(VALID_LINK);
+        assertEquals(expectedLink, ParserUtil.parseLink(VALID_LINK));
+    }
+
+    @Test
+    public void parseLink_validValueWithWhitespace_returnsTrimmedLink() throws Exception {
+        String linkWithWhitespace = WHITESPACE + VALID_LINK + WHITESPACE;
+        Link expectedLink = new Link(VALID_LINK);
+        assertEquals(expectedLink, ParserUtil.parseLink(linkWithWhitespace));
+    }
```
Comment (fy17ohhh) [inline src/test/java/educonnect/logic/parser/ParserUtilTest.java:349]: Good test coverage here

**Review by fy17ohhh — APPROVED** [review]
LGTM, just modify the comment to follow the policy on code reuse

**Review by marclamp — COMMENTED** [review]
LGTM :)

File: src/main/java/educonnect/logic/commands/HelpCommand.java [hunk 4]
```diff
@@ -1,21 +1,107 @@
...
+            + " and clears all existing tags.";
+
+    public static final String SHOWING_HELP_MESSAGE_FIND = "find - Find students whose criteria match any of "
+            + "the given keywords.\n\n"
+            + "Format: find [n/NAME] [s/STUDENT_ID] [h/TELEGRAM_HANDLE] [t/TAG]\n\n"
+            + "Example:\n"
+            + "1. find n/John returns john and John Doe\n"
+            + "2. find e/yahoo returns Alex Yeoh and David Li (both uses yahoo email)";
+
+    public static final String SHOWING_HELP_MESSAGE_CLEAR = "clear - Clears all entries from the address book.";
```
Comment (matthewken19) [inline src/main/java/educonnect/logic/commands/HelpCommand.java:62]: I noticed you've got a message directly in this file. How about importing it from our static class instead? It'll tidy things up and make future updates easier.

The message can instead be something like
`public static final String SHOWING_HELP_MESSAGE_ADD = AddCommand.MESSAGE_USAGE;`

File: src/test/java/educonnect/logic/parser/HelpCommandParserTest.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        assertParseSuccess(parser, validArgs, new HelpCommand(validArgs));
+    }
+
+    @Test
+    public void parse_unidentifiedArgument_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE);
+        assertParseFailure(parser, INVALID_ARGS, expectedMessage);
+    }
+
+}
```
Comment (matthewken19) [inline src/test/java/educonnect/logic/parser/HelpCommandParserTest.java:33]: Good test coverage!

**Review by matthewken19 — COMMENTED** [review]
Overall LGTM. Just some minor improvement to make future updates easier.

File: src/main/java/educonnect/logic/commands/HelpCommand.java (hunk 4 above)
Comment (tanjiajiajun) [reply src/main/java/educonnect/logic/commands/HelpCommand.java:62]: Good idea, i will standardise with the UG's usage guide as well. Thanks!

**Comment by darryl-chan** [comment]
LGTM

**Pull request merged**